src/cell/unsafe_cell.rs

```diff
@@ -0,0 +1,62 @@
+use crate::rt;
+
+/// A checked version of `std::cell::UnsafeCell`.
+#[derive(Debug)]
+pub struct UnsafeCell<T> {
```

Comment:
I'm not sure how I feel about this renaming. I feel like if the type shares the name with `std`'s `UnsafeCell`, users will probably expect it to have the same APIs, and it doesn't (and can't).

Reply:
Reasoning is:

- Atomic is now a "cell" and handles similar checks as `CausalCell` did.
- "CausalCell` stores an unsafe pointer, same as `UnsafeCell` and is intended to be used as the "loom" version of `UnsafeCell`.
- Other types provide different APIs than std in order to verify (see atomic types).